### ISSUE TITLE
Need to select new controller as output device before being able to configure controller mappings

### ISSUE THREAD

**Issue by spotlesscoder** [mixxxdj/mixxx#13476]
### Bug Description

I connected a DDJ-FLX4 for the first time to my macbook, allowed the access to the USB device
Then I started mixxx, went to the settings dialog. Not a single item was available under the "Controllers" node in the settings view selection

Spent a lot of time of over and over again rebooting my macbook and reconnecting the controller because I thought the USB device hadn't been registered successfully

Then I selected the DDJ-FLX4 as the audio output and suddenly the controller showed up in the Controllers list and I was finally able to configure the input mappings

I would expect that I can configure the controller mappings before selecting the device as audio device

### Version

2.4.1

### OS

macOS 14.5 (M1)

**Comment by Swiftb0y** [comment]
I faintly remember something like that on a relative's macbook. I'm fairly certain this is a macOS issue but I'm not qualified enough to look into this. Maybe @fwcd has time and interest?

**Comment by fwcd** [comment]
I can't remember running into something like that, but I also usually select my controller as output device before using it with a mapping, so I would have to test that once I find some time.

Also something I generally do (both on macOS and Linux) is making sure that the controller is plugged in (and powered) before starting Mixxx, but I think that is due to the lack of support for hot-plugging (#5614).

**Comment by Swiftb0y** [comment]
yes, that is definitely still a requirement. Don't start mixxx before the hardware is connected, it will not detect the midi endpoints correctly if afterwards (audio it can by pressing "Query Devices").

**Comment by spotlesscoder** [comment]
well I attached and approved the device before starting mixxx though
